verifier/verifier_test.go

```diff
@@ -605,3 +605,84 @@ func TestVerifyX509TrustedIdentities(t *testing.T) {
 		})
 	}
 }
+
+func TestPluginVersionCompatibility(t *testing.T) {
+	var semverSigEnv1 []byte
+	var semverSigEnv2 []byte
+	var semverSigEnv3 []byte
+	var semverSigEnv4 []byte
+	var semverSigEnv5 []byte
+	var semverSigEnv6 []byte
+
+	semverSigEnv1 = mock.MockCaMinVerSigEnv1
+	semverSigEnv2 = mock.MockCaMinVerSigEnv2
+	semverSigEnv3 = mock.MockCaMinVerSigEnv3
+	semverSigEnv4 = mock.MockCaMinVerSigEnv4
+	semverSigEnv5 = mock.MockCaMinVerSigEnv5
+	semverSigEnv6 = mock.MockCaMinVerSigEnv6
+
+	policyDocument := dummyPolicyDocument()
+
+	v := verifier{
+		trustPolicyDoc: &policyDocument,
+		pluginManager:  mock.PluginManager{},
+	}
+	opts := notation.VerifyOptions{ArtifactReference: mock.SampleArtifactUri, SignatureMediaType: "application/jose+json"}
+	envelopeMediaType := opts.SignatureMediaType
+	trustPolicy, _ := v.trustPolicyDoc.GetApplicableTrustPolicy(mock.SampleArtifactUri)
+	pluginConfig := opts.PluginConfig
+	verificationLevel, _ := trustPolicy.SignatureVerification.GetVerificationLevel()
+
+	testSignatureArrHigh := [][]byte{semverSigEnv1, semverSigEnv2, semverSigEnv3}
+	testSignatureArrLow := [][]byte{semverSigEnv4, semverSigEnv5, semverSigEnv6}
+
+	for _, testSemSigEnv := range testSignatureArrHigh {
+		outcome := &notation.VerificationOutcome{
+			RawSignature:      testSemSigEnv,
+			VerificationLevel: verificationLevel,
+		}
+		err := v.processSignature(context.Background(), testSemSigEnv, envelopeMediaType, trustPolicy, pluginConfig, outcome)
+		verificationPluginMinVersion, _ := getVerificationPluginMinVersion(&outcome.EnvelopeContent.SignerInfo)
+		outcome.Error = err
+		var errMsg string = "found plugin io.cncf.notary.plugin.unittest.mock with version 1.0.0 but signature verification needs plugin version greater than or equal to " + verificationPluginMinVersion
```

Comment:
will we use the variable if test doesnt fails?

Reply:
Modified this function and we are not using this variable anymore